src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs

```diff
@@ -847,7 +847,7 @@ public static unsafe bool TryEncode(ReadOnlySpan<char> utf16, Span<byte> utf8, o
                     int chd;
                     if (ch <= 0x7FF)
                     {
-                        if (pTarget >= pAllocatedBufferEnd - 1)
+                        if (pAllocatedBufferEnd - pTarget <= 1)
```

Comment:
Ignore this. I was thinking that we already wrote a byte and only needed space for 1 more, but we actually need space for 2 more.